Content.Client/_DV/CosmicCult/UI/Monument/MonumentMenu.xaml.cs

```diff
@@ -0,0 +1,190 @@
+        }
+
+        var entropyToNextStage = Math.Max(state.TargetProgress - state.CurrentProgress, 0);
+        var min = entropyToNextStage == 0 ? 0 : 1; //I have no idea what to call this. makes it so that it shows 0 crew for the final stage but at least one at all other times
+        var crewToNextStage = (int)Math.Max(Math.Round((double)entropyToNextStage / _ccEntropyValue, MidpointRounding.ToPositiveInfinity), min); //force it to be at least one
+
+        AvailableEntropy.Text = Loc.GetString("monument-interface-entropy-value", ("infused", availableEntropy));
+        EntropyUntilNextStage.Text = Loc.GetString("monument-interface-entropy-value", ("infused", entropyToNextStage.ToString()));
+        CrewToConvertUntilNextStage.Text = crewToNextStage.ToString();
+    }
+
+    // Update all the glyph buttons
+    private void UpdateGlyphs()
+    {
+        GlyphContainer.RemoveAllChildren();
+        foreach (var glyph in _glyphPrototypes)
+        {
+            var boxContainer = new BoxContainer();
+            var unlocked = _unlockedGlyphProtoIds.Contains(glyph.ID);
+            var button = new Button
+            {
+                HorizontalExpand = true,
+                StyleClasses = { StyleBase.ButtonSquare },
+                ToolTip = Loc.GetString(glyph.Tooltip),
+                Group = _glyphButtonGroup,
+                Pressed = glyph.ID == _selectedGlyphProtoId,
+                Disabled = !unlocked,
+                Modulate = !unlocked ? Color.Gray : Color.White,
+            };
+            button.OnPressed += _ => _selectedGlyphProtoId = glyph.ID;
+            var glyphIcon = new TextureRect
+            {
+                Texture = _sprite.Frame0(glyph.Icon),
+                TextureScale = new Vector2(2f, 2f),
+                Stretch = TextureRect.StretchMode.KeepCentered,
+            };
+            boxContainer.AddChild(button);
+            button.AddChild(glyphIcon);
+            GlyphContainer.AddChild(boxContainer);
+        }
+    }
+
+    // Update all the influence thingies
+    private void UpdateInfluences(MonumentBuiState state)
+    {
+        InfluencesContainer.RemoveAllChildren();
+
+        var influenceUIBoxes = new List<InfluenceUIBox>();
+        foreach (var influence in _influencePrototypes)
+        {
+            var uiBoxState = GetUIBoxStateForInfluence(influence, state);
+            var influenceBox = new InfluenceUIBox(influence, uiBoxState);
+            influenceUIBoxes.Add(influenceBox);
+            influenceBox.OnGainButtonPressed += () => OnGainButtonPressed?.Invoke(influence.ID);
+        }
+
+        //sort the list of UI boxes by state (locked -> owned -> not enough entropy -> enough entropy)
+        //then sort alphabetically within those categories
+        influenceUIBoxes = influenceUIBoxes.OrderBy(box => box.State)
+            .ThenBy(box => box.Proto.ID)
+            .ToList();
+
+        foreach (var box in influenceUIBoxes)
```

Comment:
Avoid double-iteration

```suggestion
        foreach (var box in influenceUIBoxes.OrderBy(box => box.State).ThenBy(box => box.Proto.ID))
```